site/src/pages/UserSettingsPage/NotificationsPage/NotificationsPage.tsx

```diff
@@ -185,6 +194,20 @@ const NotificationsPage: FC = () => {
 																			[tmpl.id]: !checked,
 																		},
 																	});
+
+																	// Clear the Tasks page warning dismissal when enabling a task notification
+																	// This ensures that if the user disables task notifications again later,
+																	// they will see the warning alert again.
+																	if (
+																		isTaskNotification(tmpl) &&
+																		checked &&
+																		preferencesQuery.data
+																	) {
+																		updatePreferencesMutation.mutate({
+																			task_notification_alert_dismissed: false,
+																		});
+																	}
```

Comment:
Fine for now, but should we consider doing this on the backend? Ideally this change would be propagated even if done directly via API or UI (or CLI if we support that, or will support that in the future).

Reply:
Yes, I also thought about that and that might be a better approach. I will address that in a follow-up PR.